File: call-in-script/1.rdseed.pl
#!/usr/bin/env perl
use strict;
use warnings;

@ARGV == 1 or die "Usage: perl $0 dirname\n";
my ($dir) = @ARGV;

chdir $dir;  # cd进数据所在目录，以避免处理路径

my @seed = glob "*.seed";
# rdseed一次只能处理一个SEED文件
die "One and only one SEED file is needed!\n" if @seed != 1;
system "rdseed -pdf @seed";

chdir "..";

File: call-in-script/1.rdseed.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
import sys
import glob
import subprocess

if len(sys.argv) != 2:
    sys.exit("Usage: python %s dirname" % sys.argv[0])

dir = sys.argv[1]

os.chdir(dir)

seed = glob.glob("*.seed")
if len(seed) != 1:
    print("One and only one SEED file is needed!")

subprocess.call(['rdseed', '-pdf', seed[0]])

os.chdir("..")
